### DISCUSSION TITLE
Frontend implementation V2

### DISCUSSION THREAD

**Discussion by caioricciuti** [caioricciuti/ch-ui#22] [General]
@thiagodsti I've created a discussion so we have a better way to discuss :) 

#21 

Ahhh cool, so really should look into that... 

There's something though... I'm referring to using Local Storage, the thing is the backend works with JWT, and in all requests it sends the user back, and also creates a `req.user` on the server which I can use to manage authorization and so on... So in theory the stores are not quite necessary for the user, I'm kinda of worried about something though, when I `console.log(any.store)` I get multiple calls to the server (I understand that when in dev mode it's duplicated) but I have like 4 to 8 every time...

**Comment by caioricciuti** [comment]
<img width="611" alt="Screenshot 2024-09-22 at 10 30 59" src="https://github.com/user-attachments/assets/1316974e-6e82-4b4a-a442-01c715aaed9a">

Example

**Reply by thiagodsti** [reply]
Aha ok it's because the useEffect and the method as deps I believe, I'll give it a look today

**Reply by thiagodsti** [reply]
Do you want to pair on it today? I'll be available at 16:00 (CEST time)

**Comment by thiagodsti** [comment]
Since it's a store here:
```
useEffect(() => {
    if (databaseData.length === 0) {
      fetchDatabaseData();
    }
  }, [fetchDatabaseData, databaseData.length]);
```

Try to change for this:
```
useEffect(() => {
    if (databaseData.length === 0) {
      fetchDatabaseData();
    }
  }, [databaseData.length]);
```

The `lint` will complain, because we don't have everything we are using inside the useEffect as deps but it's fine.

**Comment by thiagodsti** [comment]
Ahh maybe you could create a slack for this project ? It will be easier to talk there

**Reply by caioricciuti** [reply]
Hey, yep I'll create the Slack asap, it's much better, thank you for the suggestion... Another thing I've made a bunch of changes, I'll push them right now, still need to figure some things out but in theory it's better :)

**Reply by caioricciuti** [reply]
https://ch-ui-v2.slack.com/ Can you try I'm not sure how slack works actually just had an old workspace and renamed...

**Reply by thiagodsti** [reply]
You need to invite me to the slack, for example this screen you should click `Invite people to Stockholm Devs` probably it will me something similar for you.

<img width="281" alt="Screenshot 2024-09-23 at 09 25 26" src="https://github.com/user-attachments/assets/b8e4a625-2aed-481b-98aa-be7f9c8eeab0">

Also invite using the link, and you will need to change the link configuration to never expires

**Reply by caioricciuti** [reply]
Sorry for the delay, here you go! https://join.slack.com/t/ch-ui-v2/shared_invite/zt-2r6xwoizm-PMlCH6ZZAi5hK3fPagypQw

**Comment by caioricciuti** [comment]
We have opted to keep V1 rather than complicate the hole data flow. We are now using everything ClickHouse offers out of the box, no need for a separate backend since we have the best DB already available...
